common/persistence/versionhistory/version_history.go

```diff
@@ -176,6 +176,19 @@ func IsVersionHistoryItemsInSameBranch(versionHistoryItemsA []*historyspb.Versio
 	return lowestCommonAncestor.Equal(aLastItem) || lowestCommonAncestor.Equal(bLastItem)
 }
 
+func SplitVersionHistoryByLastLocalGeneratedItem(
+	versionHistoryItems []*historyspb.VersionHistoryItem,
+	initialFailoverVersion int64,
+	failoverVersionIncrement int64,
+) ([]*historyspb.VersionHistoryItem, []*historyspb.VersionHistoryItem) {
+	for i := len(versionHistoryItems) - 1; i >= 0; i-- {
+		if versionHistoryItems[i].Version%failoverVersionIncrement == initialFailoverVersion {
+			return versionHistoryItems[:i+1], versionHistoryItems[i+1:]
```

Comment:
just curious why +1, it seems like it doesn't match the function name?

Reply:
It iterate backwards to find the last local generated item. The `versionHistoryItems[:i+1]` is `[inclusive,exclusive]` while I need to include the last local generated item.

Reply:
Thank you. I understand it now